### PR TITLE
Integrate team search in Start UI

### PR DESCRIPTION
**Pull request by typfel**
This PR integrate the new `SearchDirectory` implementation along with support for teams. Some things break which I'll fix in follow up PRs.

Broken things:
* Analytics (Shumeng will discuss relevant events on friday)
* No result message

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m [hunk 1]
```diff
@@ -20,20 +20,26 @@
...
 @property (nonatomic, strong, readwrite) TokenField *tokenField;
 @end
 
 
 
 @implementation PeopleInputController
 
+- (void)dealloc
+{
+    [self.userSelection removeWithObserver:self];
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m]: Should it be `-removeObserver:` ?

File: Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m [hunk 2]
```diff
@@ -102,6 +108,13 @@ - (NSString *)plainTextContent
 
 #pragma mark - Public Interface
 
+- (void)setUserSelection:(UserSelection *)userSelection
+{
+    _userSelection = userSelection;
+    
+    [self.userSelection addWithObserver:self];
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m]: Should it be `-addObserver:`?

File: Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m [hunk 3]
```diff
@@ -128,14 +141,34 @@ - (BOOL)userDidConfirmInput
...
 - (void)tokenField:(TokenField *)tokenField changedTokensTo:(NSArray *)tokens
 {
-    if ([self.selectionDelegate respondsToSelector:@selector(peopleInputController:changedPresentedDirectoryResultsTo:)]) {
-        NSArray *users = [tokens valueForKeyPath:@"@distinctUnionOfObjects.representedObject"];
-        [self.selectionDelegate peopleInputController:self changedPresentedDirectoryResultsTo:[NSSet setWithArray:users]];
-    }
+    NSArray *users = [tokens valueForKeyPath:@"@distinctUnionOfObjects.representedObject"];
+    [self.userSelection replace:users];
+    
+//    if ([self.selectionDelegate respondsToSelector:@selector(peopleInputController:changedPresentedDirectoryResultsTo:)]) {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m]: Commented code.

File: Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m (hunk 1 above)
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/StartUI/Common/InputField/PeopleInputController.m]: Yes, forgot to add a `objc()` name

**Pull request merged**